p2p/protocol/identify/obsaddr.go

```diff
@@ -42,23 +45,40 @@ func (oa *ObservedAddr) activated(ttl time.Duration) bool {
 	return len(oa.SeenBy) >= ActivationThresh
 }
 
+type newObservation struct {
+	observed, local, observer ma.Multiaddr
+	direction                 net.Direction
+}
+
 // ObservedAddrSet keeps track of a set of ObservedAddrs
 // the zero-value is ready to be used.
 type ObservedAddrSet struct {
-	sync.Mutex // guards whole datastruct.
+	sync.RWMutex // guards whole datastruct.
 
 	// local(internal) address -> list of observed(external) addresses
 	addrs map[string][]*ObservedAddr
 	ttl   time.Duration
+
+	// this is the worker channel
+	wch chan newObservation
+}
+
+func NewObservedAddrSet(ctx context.Context) *ObservedAddrSet {
+	oas := &ObservedAddrSet{
+		addrs: make(map[string][]*ObservedAddr),
+		ttl:   pstore.OwnObservedAddrTTL,
+		wch:   make(chan newObservation, 1),
```

Comment:
Let's make this longer, like 64 or something (or at least 16). We want as many observed addrs as we can on initial start.

Reply:
Ok, will set to 16 (= 4 * activation-threshold)